pyo3-derive-backend/src/defs.rs

```diff
@@ -7,6 +7,21 @@ pub struct Proto {
     pub py_methods: &'static [PyMethod],
 }
 
+impl Proto {
+    pub(crate) fn get_proto<Q>(&self, query: Q) -> Option<&'static MethodProto>
+    where
+        Q: PartialEq<&'static str>,
```

Comment:
I think this might work without needing to be a reference?

```suggestion
        Q: PartialEq<str>,
```

Reply:
It takes `proc_macro2::Ident` as an argument, which implements `PartialEq<AsRef<str>>`.

Reply:
`str` implements `AsRef<str>` ? Maybe I'm missing something.

Reply:
`PartialEq::eq` takes `&Other` so at last it needs `&T where T: AsRef<str>`.

Reply:
Sorry, yes, I'm having a brain misfire! 🤦‍♂